lib/Conversion/TritonGPUToLLVM/TritonGPUToLLVM.cpp

```diff
@@ -3253,20 +3286,23 @@ void populateTritonToLLVMPatterns(mlir::LLVMTypeConverter &typeConverter,
                                         benefit);
   patterns.add<ArithConstantSplatOpConversion>(typeConverter, benefit);
   patterns.add<AsyncWaitOpConversion>(typeConverter, benefit);
-  patterns.add<BinaryOpConversion<arith::AddIOp, LLVM::AddOp>>(typeConverter,
+  patterns.add<ElementwiseOpConversion<arith::AddIOp, LLVM::AddOp>>(typeConverter,
+                                                               benefit);
+  patterns.add<ElementwiseOpConversion<arith::SubIOp, LLVM::SubOp>>(typeConverter,
                                                                benefit);
-  patterns.add<BinaryOpConversion<arith::AddFOp, LLVM::FAddOp>>(typeConverter,
+  patterns.add<ElementwiseOpConversion<arith::AddFOp, LLVM::FAddOp>>(typeConverter,
                                                                 benefit);
-  patterns.add<BinaryOpConversion<arith::MulIOp, LLVM::MulOp>>(typeConverter,
+  patterns.add<ElementwiseOpConversion<arith::SubFOp, LLVM::FSubOp>>(typeConverter,
+                                                                benefit);
+  patterns.add<ElementwiseOpConversion<arith::MulIOp, LLVM::MulOp>>(typeConverter,
                                                                benefit);
-  patterns.add<BinaryOpConversion<arith::MulFOp, LLVM::FMulOp>>(typeConverter,
+  patterns.add<ElementwiseOpConversion<arith::MulFOp, LLVM::FMulOp>>(typeConverter,
                                                                 benefit);
 
-  patterns.add<BinaryOpConversion<arith::AndIOp, LLVM::AndOp>>(typeConverter,
+  patterns.add<ElementwiseOpConversion<arith::AndIOp, LLVM::AndOp>>(typeConverter,
                                                                benefit);
-  patterns.add<BinaryOpConversion<arith::OrIOp, LLVM::OrOp>>(typeConverter,
+  patterns.add<ElementwiseOpConversion<arith::OrIOp, LLVM::OrOp>>(typeConverter,
```

Comment:
Have we added all the needed elementwise in this PR or plan them in future PRs?
"select" is an example of ternary elemwise ops i can think of, and it seems still missing here.

Reply:
I don't add all elementwise ops in this MR. This MR is mainly for external libs.